apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -128,6 +115,31 @@ where
         }
     }
 
+    /// Calculates what the [`TxResult`]s would be for the transactions in a
+    /// proposed block, as well as counting the number of digest transactions
+    /// present. `ProcessProposal` should be able to make a decision on whether
+    /// a proposed block is acceptable or not based solely on what this function
+    /// returns.
```

Comment:
```suggestion
    /// Calculates what the [`TxResult`]s would be for the transactions in a
    /// proposed block, and counts the number of digest transactions
    /// present. 
```
leave the last part to `vote_on_proposal`, with a bit of clean up:
```
    /// Decides whether `ProcessProposal` should accept or reject
    /// a block proposed by some Tendermint round's leader.
    ///
    /// We first call `check_tx_status()`, to check for non-recoverable
    /// transaction errors.
```

Reply:
unsure about this suggestion, if we don't create a `vote_on_proposal` fn in this PR